File: Resueltos_por_competidor/Miguel_Lopez/Magnets.cpp
/*
 * Autor: Miguel Lopez
 * Problema: Magnets
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/344/A
 */

#include <bits/stdc++.h>
#include <cctype>
#include <cstdio>
typedef long long int ll;
using namespace std;
void init_code() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    #endif 
}
int main() {
   init_code();
   int t, cont = 0; cin >> t;
   string a,b;
   for(int i = 0; i < t ; i++) {
      cin >> b;
      if(i == 0 || a != b) {
         cont++;
      }
      a = b;
      
   }
   cout << cont << endl;
   return 0;
}

File: Resueltos_por_competidor/Gabriella_Castro/Magnets.java
/*
 * Autor: Gabriella Castro
 * Problema: Magnets
 * Juez online: Codeforces
 * Veredicto: Accepted
 * URL: https://codeforces.com/problemset/problem/344/A
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Magnets {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine().trim());
        
        int grupos = 1;
        String previo = br.readLine().trim();
        
        for (int i = 1; i < n; i++) {
            String actual = br.readLine().trim();
            if (!actual.equals(previo)) {
                grupos++;
                previo = actual;
            }
        }
        
        System.out.println(grupos);
    }
}
